File: 2026_GFG_POTD/JANUARY/06-01-2026/Max Xor Subarray of size K.cs
public class Solution {
    public int maxSubarrayXOR(int[] arr, int k) {
        int n = arr.Length;
        if (k > n) return 0;
        
        int xorSum = 0;
        for (int i = 0; i < k; i++)
            xorSum ^= arr[i];
        
        int maxXor = xorSum;
        
        for (int i = k; i < n; i++) {
            xorSum ^= arr[i] ^ arr[i - k];
            if (xorSum > maxXor) maxXor = xorSum;
        }
        
        return maxXor;
    }
}

File: 2026_GFG_POTD/JANUARY/06-01-2026/Max Xor Subarray of size K.js
class Solution {
    /**
     * @param {number[]} arr
     * @param {number} k
     * @returns {number}
     */
    maxSubarrayXOR(arr, k) {
        const n = arr.length;
        if (k > n) return 0;

        let xorSum = 0;
        for (let i = 0; i < k; i++) xorSum ^= arr[i];

        let maxXor = xorSum;

        for (let i = k; i < n; i++) {
            xorSum ^= arr[i] ^ arr[i - k];
            if (xorSum > maxXor) maxXor = xorSum;
        }

        return maxXor;
    }
}